pages/docs/styled-system/utility-hooks/use-dimensions.mdx

```diff
@@ -0,0 +1,89 @@
+---
+title: useDimensions
+package: '@chakra-ui/hooks'
+description: 'React hook to measure dimensions of the referenced element'
+---
+
+`useDimensions` is a custom hook that measures dimensions of the referenced
+element based on its box-model.
+
+## Import
+
+```js
+import { useDimensions } from '@chakra-ui/react'
+```
+
+## Return value
+
+This hook returns an object with the properties `marginBox`, `paddingBox`,
+`borderBox`, `conentBox`, `border`, `padding`, and `margin`.
+
+Each of these properties contains a nested object which provides values
+respective to that property:
+
+| Inside the `*Box` properties | Inside `border`, `padding`, and `margin` properties |
+| ---------------------------- | --------------------------------------------------- |
+| `top`                        | `top`                                               |
+| `right`                      | `right`                                             |
+| `bottom`                     | `bottom`                                            |
+| `left`                       | `left`                                              |
+| `width`                      |                                                     |
+| `height`                     |                                                     |
+| `x`                          |                                                     |
+| `y`                          |                                                     |
+| `center` (x,y)               |                                                     |
+
+## Usage
+
+```jsx
+function example() {
+  const elementRef = useRef()
+  const dimensions = useDimensions(elementRef)
+
+  return (
+    <Box ref={elementRef}>
```

Comment:
I think it'll be a better UX if we set the width of the `Box` to `fit-content` and add a background, so when the visitors play with the text they can see how the Box shrinks and grows, and also see the dimensions text change accordingly.

Reply:
@nikolovlazar do you think it would also help to add a note above the example to encourage editing the text?

Reply:
Don't think so. It says "EDITABLE EXAMPLE" on top of the code block.

Reply:
Sounds good!